File: src/main/java/com/blooddonation/system/models/donations/MoneyDonation/PaymentMethod/Online/Online.java
package com.blooddonation.system.models.donations.MoneyDonation.PaymentMethod.Online;
import com.blooddonation.system.models.donations.MoneyDonation.PaymentMethod.PaymentMethod;
import com.blooddonation.system.models.donations.DonationMethod;

public class Online implements DonationMethod {
    private PaymentMethod paymentMethod;

    public Online(PaymentMethod paymentMethod)
    {
        this.paymentMethod = paymentMethod;
    }

    @Override
    public boolean donate(float amount) {
        return paymentMethod.processPayment(amount);
    }
}

File: models/MoneyDonation/Online/Online.php
<?php
require_once $_SERVER['DOCUMENT_ROOT'] . '/services/database_service.php';

class Online implements IMoneyDonationMethod {
    private IPaymentMethod $paymentMethod;

    public function __construct($paymentMethod) {
        $this->paymentMethod = $paymentMethod;
    }

    public function donate($amount): bool {
        $this->paymentMethod->processPayment($amount);
        return true;
    }
}